vortex-array/src/compute/scalar_subtract.rs

```diff
@@ -0,0 +1,36 @@
+use vortex_dtype::DType;
+use vortex_error::{vortex_err, VortexResult};
+
+use crate::scalar::Scalar;
+use crate::{Array, ArrayDType, OwnedArray};
+
+pub trait ScalarSubtractFn {
+    fn scalar_subtract(&self, to_subtract: &Scalar) -> VortexResult<OwnedArray>;
+}
+
+pub fn scalar_subtract<T: Into<Scalar>>(array: &Array, to_subtract: T) -> VortexResult<OwnedArray> {
```

Comment:
I might have been wrong about taking `T: Into<Scalar>` here, since it forces you to alloc/clone scalars during the recursive call.

Reply:
Yeah the into makes for a nicer API, but I can't figure out how to avoid the clone -- I'll change it back.